node/impl/full/eth.go

```diff
@@ -1741,7 +1695,8 @@ func newEthTxReceipt(ctx context.Context, tx ethtypes.EthTx, lookup *api.MsgLook
 		LogsBloom:        ethtypes.EmptyEthBloom[:],
 	}
 
-	if receipt.To == nil && lookup.Receipt.ExitCode.IsSuccess() {
+	// V.Int will be nil for non-eth transactions, so we want to skip unmarshalling the create return.
```

Comment:
I'm not sure I understand what's going on here -- why is V.Int relevant to the unmarshaling of the create return? Regardless of whether this was an eth tx or not, the return should be unmarshalable if the message was successful?

Reply:
The issue was that I didn't want the unmarshal to fail and return an empty struct. I've refactored this function to do all the general stuff first, and then return early if this isn't an eth tx.